cmd/infra/powervs/create_test.go

```diff
@@ -0,0 +1,41 @@
+package powervs
+
+import (
+	"testing"
+
+	"github.com/IBM-Cloud/power-go-client/power/models"
+)
+
+func TestUseExistingDHCP(t *testing.T) {
+	id1 := "id1"
+	id2 := "id2"
+
+	type expectedResult struct {
+		dhcpServerID string
+		err          error
+	}
+
+	tests := []struct {
+		input          models.DHCPServers
+		expectedResult expectedResult
+		description    string
+	}{
+		{
+			input:          models.DHCPServers{{ID: &id1}},
+			expectedResult: expectedResult{dhcpServerID: id1, err: nil},
+			description:    "expected DHCPServerDetail{ID: id1} to be returned with nil error",
+		},
+		{
+			input:          models.DHCPServers{{ID: &id1}, {ID: &id2}},
+			expectedResult: expectedResult{"", dhcpServerLimitExceeds(2)},
+			description:    "expected error to be returned when more than one DHCPServer exist",
+		},
+	}
+
+	for _, test := range tests {
+		dhcpServerID, err := useExistingDHCP(test.input)
+		if dhcpServerID != test.expectedResult.dhcpServerID || (err != nil && err.Error() != test.expectedResult.err.Error()) {
```

Comment:
error check can't be skipped we should always check even if there is data in the first value or not. I prefer using [stretchr/testify](https://github.com/stretchr/testify#assert-package) to have better assertion.